src/recon.rs

```diff
@@ -2648,13 +2647,16 @@ pub(crate) unsafe fn rav1d_recon_b_intra<BD: BitDepth>(
                         } else {
                             EDGE_I444_LEFT_HAS_BOTTOM as c_int
                         })) as EdgeFlags;
-                        top_sb_edge = 0 as *const BD::Pixel;
-                        if t.by & (*f).sb_step - 1 == 0 {
-                            top_sb_edge = (*f).ipred_edge[0] as *mut BD::Pixel;
+                        let top_sb_edge_slice = if t.by & (*f).sb_step - 1 == 0 {
+                            let mut top_sb_edge: *const BD::Pixel =
+                                (*f).ipred_edge[0] as *mut BD::Pixel;
                             let sby = t.by >> (*f).sb_shift;
                             top_sb_edge =
                                 top_sb_edge.offset(((*f).sb128w * 128 * (sby - 1)) as isize);
-                        }
+                            slice::from_raw_parts(top_sb_edge, (*f).ipred_edge_sz as usize * 128)
```

Comment:
Is `ipred_edge_sz` the full length of the slice `ipred_edge[0]`, or the length left after offsetting by `sb128w * 128 * (sby - 1)`?

Reply:
`ipred_edge_sz * 128` is the full length in bytes. Some fixing is definitely needed here.